test/fuzz/pem/fuzzcert_test.go

```diff
@@ -84,3 +86,94 @@ func FuzzUnmarshalPEMToPublicKey(f *testing.F) {
 		}
 	})
 }
+
+func FuzzCertificate(f *testing.F) {
+	f.Fuzz(func(t *testing.T, withLimited bool, pemBytes []byte, iterations int) {
+		var certs []*x509.Certificate
+		var err error
+		if withLimited {
+			certs, err = cryptoutils.UnmarshalCertificatesFromPEMLimited(pemBytes, iterations)
+			if err != nil {
+				return
+			}
+			cryptoutils.MarshalCertificatesToPEM(certs)
+		} else {
+			certs, err = cryptoutils.UnmarshalCertificatesFromPEM(pemBytes)
+			if err != nil {
+				return
+			}
+			cryptoutils.MarshalCertificatesToPEM(certs)
+		}
+		for _, cert := range certs {
+			cryptoutils.GetSubjectAlternateNames(cert)
+		}
+	})
+}
+
+func FuzzPrivateKey(f *testing.F) {
+	f.Fuzz(func(t *testing.T, pemBytes, password []byte) {
+		passFunc := cryptoutils.StaticPasswordFunc(password)
+		privateKey, err := cryptoutils.UnmarshalPEMToPrivateKey(pemBytes, passFunc)
+		if err != nil {
+			return
+		}
+
+		cryptoutils.MarshalPrivateKeyToPEM(privateKey)
+		cryptoutils.MarshalPrivateKeyToEncryptedDER(privateKey, passFunc)
+	})
+}
+
+func FuzzParseCSR(f *testing.F) {
```

Comment:
I'm iffy on if we need this one, because it's just a call to `pem.Decode` and `x509.ParseCertificateRequest` with some conditionals. I suspect we'll mostly see timeouts on this fuzzer, so I'd probably remove.